### DISCUSSION TITLE
Item Remove

### DISCUSSION THREAD

**Discussion by sarathdm** [moudey/Shell#122] [General]
How to remove this item
![AXJCLv9hTD](https://user-images.githubusercontent.com/117918597/213845925-b54baff9-8db5-48ee-b633-be73900cbd68.png)

**Comment by moudey** [comment]
Open the /imports/terminal.nss file located in Shell directory.
replace `title.terminal`  with `id.terminal.title`, and then restart File Explorer from Shell.exe

**Comment by sarathdm** [comment]
Icon misplaced and no title showing
![explorer_o7oUQTHC56](https://user-images.githubusercontent.com/117918597/213847439-a1a21049-2bcf-4d60-8404-3a1f10b4eb67.png)

**Comment by moudey** [comment]
Open `/imports/terminal.nss` replace `title.windows_terminal` with `id.windows_terminal.title`

**Comment by sarathdm** [comment]
thanks

**Comment by sarathdm** [comment]
Any way to remove or hide these from this context menu
![br9XUMS8bK](https://user-images.githubusercontent.com/117918597/213848077-1e3c89e3-6fdb-42bf-be04-fea486e5bf3a.png)

**Comment by moudey** [comment]
Open `shell.nss` and add this code to `static` section

```
item(type='taskbar' find='toolbar' vis=vis.remove)
item(type='taskbar' find='show cortana' vis=vis.disable)
//etc...
```

**Comment by sarathdm** [comment]
ok

**Comment by sarathdm** [comment]
Icon misplaced
![WSorWa70lJ](https://user-images.githubusercontent.com/117918597/213849368-330d9e53-cd6f-4300-b511-ae584467b786.png)

**Comment by moudey** [comment]
Choose one from this [link](https://nilesoft.org/gallery/glyphs)

Open `shell.nss` and add this code to `static` section
```
item(type='taskbar' find='lock the taskbar' image=\uE19A)
```

**Comment by sarathdm** [comment]
ok

**Comment by sarathdm** [comment]
thanks

**Comment by sarathdm** [comment]
remove item from this context menu only
![eJlfTDPzfv](https://user-images.githubusercontent.com/117918597/213862811-401185b9-9497-4517-b0ac-e766a4b2475a.png)

**Comment by sarathdm** [comment]
Remove form the menu and add icons
![Gng2GtvrxC](https://user-images.githubusercontent.com/117918597/213877460-1b1ba1ce-56b0-47aa-8f62-3d71671c4483.png)

**Reply by moudey** [reply]
Please refer to this topic #117

**Reply by moudey** [reply]
Not all items have an icon

**Reply by sarathdm** [reply]
ok

**Comment by sarathdm** [comment]
item goto remove from menu
![KAX3u8FZbZ](https://user-images.githubusercontent.com/117918597/213901501-6f23e244-ed0b-4a23-87f3-ee9a24231bdf.png)

i typed item(type='*' where=window.is_start find='goto' vis=vis.remove) in shell.nss
but not working......
any idea to remove goto in that menu

**Reply by moudey** [reply]
"Go to" is a dynamic item we just need to change some properties

Open `/imports/goto.nss` replace the value of `where` property with the following
```
where=!window.is_start && (window.is_taskbar||sel.count)
```

**Reply by sarathdm** [reply]
ok
Thanks

**Comment by iamLazyCode** [comment]
hi sir how do i remove this item
![image](https://user-images.githubusercontent.com/13099434/213933571-564bb403-d270-49c6-9afd-323b99fe9106.png)

**Reply by moudey** [reply]
Open `/imports/develop.nss`, ignore the code with a multi-line comment or delete it
```
/*

menu(mode="single" title='editors' image=\uE17A)
{
	item(title='Visual Studio Code' image=[\uE272, #22A7F2] cmd='code' args='"@sel.path"')
	separator
	item(type='file' mode="single" title='Windows notepad' image cmd='@sys.bin\notepad.exe' args='"@sel.path"')
}

*/
```
Save file, hold `CTRL`+`RIGHT-CLICK` to reload settings.

**Reply by iamLazyCode** [reply]
> Open `/imports/develop.nss`, ignore the code with a multi-line comment or delete it
> 
> ```
> /*
> 
> menu(mode="single" title='editors' image=\uE17A)
> {
> 	item(title='Visual Studio Code' image=[\uE272, #22A7F2] cmd='code' args='"@sel.path"')
> 	separator
> 	item(type='file' mode="single" title='Windows notepad' image cmd='@sys.bin\notepad.exe' args='"@sel.path"')
> }
> 
> */
> ```
> 
> Save file, hold `CTRL`+`RIGHT-CLICK` to reload settings.

thankyou ,, what if i want to delete this whole section develop ? also i use different method of implementing terminal in right click how di remove that all together

**Reply by moudey** [reply]
Open `/shell.nss', delete this line
```
import 'imports/develop.nss'
```
Save file, hold `CTRL`+`RIGHT-CLICK` to reload settings.

**Reply by iamLazyCode** [reply]
thankyou sir one coffee coming up .... also how i make a sub menu and keep all my desired apps under one section

like i want to make 'A' and when i hover on to it shows all the apps and also how i delete the specific one

![image](https://user-images.githubusercontent.com/13099434/213934808-1d047192-283a-4790-ad4f-8d131563868e.png)

**Reply by iamLazyCode** [reply]
> Open `/shell.nss', delete this line
> 
> ```
> import 'imports/develop.nss'
> ```
> 
> Save file, hold `CTRL`+`RIGHT-CLICK` to reload settings.

not much but a coffee on my behalf
![image](https://user-images.githubusercontent.com/13099434/213935040-567045b9-fdec-4a17-989a-4a86c42606ce.png)

**Reply by moudey** [reply]
Open `shell.nss`

First, create an item of menu type in `dynamic` section, for example below
```
menu(title="A" image=#0f0)
{
   item(title='test item' image=[\uE1A8, #f00] cmd=msg('hello '+ user.name))
}
```

Second, create an item in `static` section that searches and moves to the submenu, for example below
```
item(find='git gui here|git bash here' menu="a")
item(find='treesize free' pos=0 sep='bottom' menu="a")
```

Save file, hold `CTRL`+`RIGHT-CLICK` or restart Explorer by `shell.exe -restart` to reload settings.

**Reply by iamLazyCode** [reply]
> Open `shell.nss`
> 
> First, create an item of menu type in `dynamic` section, for example below
> 
> ```
> menu(title="A" image=#0f0)
> {
>    item(title='test item' image=[\uE1A8, #f00] cmd=msg('hello '+ user.name))
> }
> ```
> 
> Second, create an item in `static` section that searches and moves to the submenu, for example below
> 
> ```
> item(find='git gui here|git bash here' menu="a")
> item(find='treesize free' pos=0 sep='bottom' menu="a")
> ```
> 
> Save the file, hold `CTRL`+`RIGHT-CLICK` or restart Explorer by `shell.exe -restart` to reload settings.

lemme try sir :P


absolutely amazing sir
![image](https://user-images.githubusercontent.com/13099434/213936220-788eeb3f-ad7d-4fdc-abdb-364a5bdf6b96.png)


 how do I add any script /app that's located on windows but not in the right menu for quick access?

**Reply by moudey** [reply]
To add a program to the context menu, add new item in `dynamic` section, see the following example
```
dynamic
{
    item(title="Regedit" image cmd='regedit.exe')
    item(title="my notepad" image cmd='notepad.exe' menu="a")
}
```

**Reply by iamLazyCode** [reply]
> To add a program to the context menu, add new item in `dynamic` section, see the following example
> 
> ```
> dynamic
> {
>     item(title="Regedit" image cmd='regedit.exe')
>     item(title="my notepad" image cmd='notepad.exe' menu="a")
> }
> ```

thanks sir please can u post command to add program or script i have kept in drive d:/mystuff/scripts/mydata.bat

**Reply by iamLazyCode** [reply]
> To add a program to the context menu, add new item in `dynamic` section, see the following example
> 
> ```
> dynamic
> {
>     item(title="Regedit" image cmd='regedit.exe')
>     item(title="my notepad" image cmd='notepad.exe' menu="a")
> }
> ```

also sir can you tell me how can i move this out side of  more section

![image](https://user-images.githubusercontent.com/13099434/214379128-a2c79df2-4e64-4209-b757-16685aa67eb9.png)

**Reply by iamLazyCode** [reply]
@moudey sir

**Reply by moudey** [reply]
Hi @ifelixculpa
> can you tell me how can i move this out side of more section

Moved this item because items longer than 25 characters have been migrated to another "More options" submenu.
You have two options, either exclude this item from moving or unmove items with long titles

1. exclude this item from moving
Open `shell.nss` and find this item in `static` section
```
item(where=this.title.length > 25 menu=title.more_options)
item(find="dell display" vis=0)
```
Change it to
```
item(where=(this.title.length > 25 && !str.start(this.title, "windows terminal here")) menu=title.more_options)
```
2. unmove items with long titles, 
Add // at the front of the line to ignore this command
```
// item(where=)this.title.length > 25( menu=title.more_options)
```

**Reply by iamLazyCode** [reply]
> Hi @ifelixculpa
> > can you tell me how can i move this out side of more section
> 
> Moved this item because items longer than 25 characters have been migrated to another "More options" submenu.
> You have two options, either exclude this item from moving or unmove items with long titles
> 
> 1. exclude this item from moving
> Open `shell.nss` and find this item in `static` section
> ```
> item(where=this.title.length > 25 menu=title.more_options)
> item(find="dell display" vis=0)
> ```
> Change it to
> ```
> item(where=(this.title.length > 25 && !str.start(this.title, "windows terminal here")) menu=title.more_options)
> ```
> 2. unmove items with long titles, 
> Add // at the front of the line to ignore this command
> ```
> // item(where=)this.title.length > 25( menu=title.more_options)
> ```
> 



Thanks amillion sir just last question is this sir , you sorted everything else nicely sir

thanks sir please can u post command to add program or script i have kept in drive d:/mystuff/scripts/mydata.bat

**Reply by moudey** [reply]
> can u post command to add program or script i have kept in drive d:/mystuff/scripts/mydata.bat

```
dynamic
{
    // ...
    item(title='run myscript' cmd='d:/mystuff/scripts/mydata.bat')
    // ....
}
```

**Reply by iamLazyCode** [reply]
> > can u post command to add program or script i have kept in drive d:/mystuff/scripts/mydata.bat
> 
> ```
> dynamic
> {
>     // ...
>     item(title='run myscript' cmd='d:/mystuff/scripts/mydata.bat')
>     // ....
> }
> ```

thankyou soooooo much sir

**Comment by iamLazyCode** [comment]
sir i added the script which runs fine when i double click it but when i add that to the right click shell extension nothing happens

**Reply by moudey** [reply]
> sir i added the script which runs fine when i double click it but when i add that to the right click shell extension nothing happens

The code should work. Is the path of the patch file correct?

To keep the patch window from closing
```
item(title='run myscript' cmd args='/k d:/mystuff/scripts/mydata.bat')
```

**Reply by iamLazyCode** [reply]
> item(title='run myscript' cmd args='/k d:/mystuff/scripts/mydata.bat')

this one worked like a charm , thanks a million for non stop help thanks alot sir
